### ISSUE TITLE
Swift 3 Support

### ISSUE THREAD

**Issue by ctews** [Haneke/HanekeSwift#329]
Hey guys,

I saw that theres a branch `feature/swift-3` but it doesn't seem to be maintained. Are there any news regarding Swift 3 support? Would be great!

**Comment by alpennec** [comment]
I am also interested.
I am not able to build the Haneke framework using Carthage with Xcode 8 b2 / Swift 3.
I use the "feature/swift-3" branch. Find below the error message logged in the terminal.

```
*** Building scheme "Haneke-iOS" in Haneke.xcworkspace
--- xcodebuild: WARNING: Unable to open project file '/Users/Axel/Developer/Skipper-iOS/Carthage/Checkouts/HanekeSwift/Haneke.playground' in workspace '/Users/Axel/Developer/Skipper-iOS/Carthage/Checkouts/HanekeSwift/Haneke.xcworkspace'.
A shell task (/usr/bin/xcrun xcodebuild -workspace /Users/Axel/Developer/Skipper-iOS/Carthage/Checkouts/HanekeSwift/Haneke.xcworkspace -scheme Haneke-iOS -configuration Release -sdk iphoneos ONLY_ACTIVE_ARCH=NO BITCODE_GENERATION_MODE=bitcode CODE_SIGNING_REQUIRED=NO CODE_SIGN_IDENTITY= CARTHAGE=YES clean build) failed with exit code 65:
--- xcodebuild: WARNING: Unable to open project file '/Users/Axel/Developer/Skipper-iOS/Carthage/Checkouts/HanekeSwift/Haneke.playground' in workspace '/Users/Axel/Developer/Skipper-iOS/Carthage/Checkouts/HanekeSwift/Haneke.xcworkspace'.
** BUILD FAILED **


The following build commands failed:
	CompileSwift normal arm64 /Users/Axel/Developer/Skipper-iOS/Carthage/Checkouts/HanekeSwift/Haneke/UIView+Haneke.swift
	CompileSwift normal arm64 /Users/Axel/Developer/Skipper-iOS/Carthage/Checkouts/HanekeSwift/Haneke/Fetch.swift
	CompileSwift normal arm64 /Users/Axel/Developer/Skipper-iOS/Carthage/Checkouts/HanekeSwift/Haneke/UIButton+Haneke.swift
	CompileSwift normal arm64 /Users/Axel/Developer/Skipper-iOS/Carthage/Checkouts/HanekeSwift/Haneke/String+Haneke.swift
	CompileSwiftSources normal arm64 com.apple.xcode.tools.swift.compiler
(5 failures)
```

<img width="1029" alt="screenshot 2016-07-14 21 23 26" src="https://cloud.githubusercontent.com/assets/2082662/16852642/3a778c68-4a09-11e6-9d09-a573988f3c70.png">

Any idea how to get a correct compilation?
Thanks,
Axel

**Comment by Tokanod** [comment]
Hi,

It's a bit worrying not to have any answer on this thread for 19 days.
Is it a summer holidays effect or there is no plan anymore to maintain HanekeSwift?

Thanks.

**Comment by Nemixdre** [comment]
Hi,

It's a bit worrying not to have any answer on this thread for 19 days.
Is it a summer holidays effect or there is no plan anymore to maintain HanekeSwift?

Thanks.

**Comment by KTMarc** [comment]
I guess that people have their own paid work and they do open source when they can.

In one of my projects I had this HanekeSwift pod and now I can't use XCode 8 (and its shiny new features) because I'm relying too much on this library. It's too much inside the core of my logic. That's because the library is really good and does many things 😌

I'm creating another branch for my project to substitute this library by my own code and be able to use Swift 3 in XCode 8. That helps me understand better how to use NSCache and at the same time how to implement a modular architecture that doesn't rely on a particular framework. 

I already removed SwiftyJSON and Alamofire which are wrappers for common Cocoa classes that everybody can use. 

Let's give them some time until the summer finishes and xcode 8 GM comes out!

**Comment by carobeta** [comment]
I made a migration to swift 3. There is in a new pull request, maybe is useful.

**Comment by lubos08** [comment]
No solution yet?
I really like this library, but I can't use with xcode 8 / swift 3.0

**Comment by jlaws** [comment]
There are multiple PR's to add Swift 3 support, can we get one of them merged please?

**Comment by JohnCoates** [comment]
I came across this error also. Personally, I'm looking at switching to a similar library, [Kingfisher](https://github.com/onevcat/Kingfisher), which seems to be pretty active

**Comment by samuel-mellert** [comment]
Is anyone still feeling responsible for keeping this project alive, or is a fork the only way to get this thing into the post Swift 2.2 era? I really like this piece of code!

**Comment by blainerothrock** [comment]
No progress on this yet? Bummer, this is the last dependency I am waiting on.. I am going to have to switch to something else.

**Comment by okankocyigit** [comment]
could you please at least inform about swift 3 support? -/+?

**Comment by dcharbonnier** [comment]
Any pr that pass the travis build will be merged.

**Comment by okankocyigit** [comment]
this [pr](https://github.com/Haneke/HanekeSwift/pull/338) is working for me.

**Comment by seriyvolk83** [comment]
If code from feature/swift-3 is used the app crash sometimes.
